deploy-tool/main.go

```diff
@@ -0,0 +1,227 @@
+// terraform/modules/facilitator/facilitator.tf and the corresponding structs in
+// facilitator/src/manifest.rs.
+
+// BatchSigningPublicKey represents a public key used for batch signing.
+type BatchSigningPublicKey struct {
+	// PublicKey is the PEM armored base64 encoding of the ASN.1 encoding of the
+	// PKIX SubjectPublicKeyInfo structure. It must be an ECDSA P256 key.
+	PublicKey string `json:"public-key"`
+	// Expiration is the ISO 8601 encoded UTC date at which this key expires.
+	Expiration string `json:"expiration"`
+}
+
+// PacketEncryptionCertificate represents a certificate containing a public key
+// used for packet encryption.
+type PacketEncryptionCertificate struct {
+	// Certificate is the PEM armored X.509 certificate.
+	Certificate string `json:"certificate"`
+}
+
+// SpecificManifest represents the manifest file advertised by a data share
+// processor. See the design document for the full specification.
+// https://docs.google.com/document/d/1MdfM3QT63ISU70l63bwzTrxr93Z7Tv7EDjLfammzo6Q/edit#heading=h.3j8dgxqo5h68
+type SpecificManifest struct {
+	// Format is the version of the manifest.
+	Format float64 `json:"format"`
+	// IngestionBucket is the region+name of the bucket that the data share
+	// processor which owns the manifest reads ingestion batches from.
+	IngestionBucket string `json:"ingestion-bucket"`
+	// PeerValidationBucket is the region+name of the bucket that the data share
+	// processor which owns the manifest reads peer validation batches from.
+	PeerValidationBucket string `json:"peer-validation-bucket"`
+	// BatchSigningPublicKeys maps key identifiers to batch signing public keys.
+	// These are the keys that peers reading batches emitted by this data share
+	// processor use to verify signatures.
+	BatchSigningPublicKeys map[string]BatchSigningPublicKey `json:"batch-signing-public-keys"`
+	// PacketEncryptionCertificates maps key identifiers to packet encryption
+	// certificates. The values are PEM encoded X.509 certificates, which
+	// contain the public key corresponding to the private key that the data
+	// share processor which owns the manifest uses to decrypt ingestion share
+	// packets.
+	PacketEncryptionCertificates map[string]PacketEncryptionCertificate `json:"packet-encryption-certificates"`
+}
+
+// TerraformOutput represents the JSON output from `terraform apply` or
+// `terraform output --json`. This struct must match the output variables
+// defined in terraform/main.tf, though it only need describe the output
+// variables this program is interested in.
+type TerraformOutput struct {
+	ManifestBucket struct {
+		Value string
+	} `json:"manifest_bucket"`
+	SpecificManifests struct {
+		Value map[string]struct {
+			KubernetesNamespace string           `json:"kubernetes-namespace"`
+			SpecificManifest    SpecificManifest `json:"specific-manifest"`
+		}
+	} `json:"specific_manifests"`
+}
+
+func generateKeyPair(namespace, keyName string) (*ecdsa.PrivateKey, error) {
```

Comment:
Don't know how I feel about this method also interfacing with kubernetes. Makes testing this code a bit harder too from what I can gather.

Reply:
You mean `os.Exec("kubectl")`? I agree, it's pretty bad. It's also pretty bad that we run this locally on a deployer's personal machine, because it exposes the keys to us individually. What I think we should do for real is run this key generation job in the k8s cluster, and as a cronjob so it can automatically rotate keys, which is tracked by #24. The goal here is to enable us to stand up an instance for Narnia and for Atredis to look at on the 26th.

However I agree the the function name could clarify that it also populates a k8s secret.

Reply:
That's fair yeah, we're taking plenty shortcuts :D

Reply:
+1 on renaming this to reflect that it also uploads the secret